### PR TITLE
cip for coordinated network upgrades

### PR DESCRIPTION
**Pull request by cmwaters**
Adds a draft CIP defining a protocol for coordinated upgrades

### CODE REVIEW

File: cips/cip-network-upgrade.md
Comment (evan-forbes) [inline cips/cip-network-upgrade.md:57]: this is really good rationale for why the percentage is over 2/3s, but doesn't contian the rest of the rational for upgrades that don't use token voting or single binaries. I'm not blocking on it, but we might want to include more here from previous documents

**Review by evan-forbes — APPROVED** [review]
:fire:

File: cips/cip-network-upgrade.md
Comment (YazzyYaz) [inline cips/cip-network-upgrade.md:5]: post a URL please from forum for discussions

File: cips/cip-network-upgrade.md
Comment (YazzyYaz) [inline cips/cip-network-upgrade.md:69]: Is there a Reference Implementation section?

File: cips/cip-network-upgrade.md
Comment (YazzyYaz) [reply cips/cip-network-upgrade.md:57]: I would recommend discussions on the proposal to happen in a forum post that @cmwaters should provide. PRs get merged so convos get lost over time.

Comment (evan-forbes) [reply cips/cip-network-upgrade.md:57]: SGTM and makes sense

File: cips/cip-network-upgrade.md
Comment (cmwaters) [reply cips/cip-network-upgrade.md:69]: They use their operator account key (not their cometbft consensus key). It is not automatic, however you could have something setup with authz that permitted the node to automatically sign the signal version transaction. I think this is outside the scope of the CIP which should be focused on the specification change to the state machine

File: cips/cip-network-upgrade.md
Comment (cmwaters) [reply cips/cip-network-upgrade.md]: Done although it's a bit of a chicken and egg problem. I will update the forum discussion to link to the actual document instead of the PR for the document after merging this

File: cips/cip-network-upgrade.md
Comment (YazzyYaz) [reply cips/cip-network-upgrade.md:69]: yes you can. I just meant having the section here, even if the implementation isn't ready. can just add to it "TBA" or something. so far, there's no Implementation section

Comment (rootulp) [reply cips/cip-network-upgrade.md:69]: Agreed out of scope. Resolving

Comment (cmwaters) [reply cips/cip-network-upgrade.md:73]: Done

**Review by YazzyYaz — APPROVED** [review]

**Pull request merged**
